modules/swagger-codegen/src/main/java/com/wordnik/swagger/codegen/examples/ExampleGenerator.java

```diff
@@ -53,12 +65,10 @@ else if(property != null && mediaType.startsWith("application/xml")) {
       }
     }
     else {
-      for(String key: examples.keySet()) {
-        String value = examples.get(key);
-
-        Map<String, String> kv = new HashMap<String, String>();
-        kv.put("contentType", key);
-        kv.put("example", value);
+      for(Map.Entry<String, Object> entry: examples.entrySet()) {
+        final Map<String, String> kv = new HashMap<String, String>();
+        kv.put("contentType", entry.getKey());
+        kv.put("example", Json.pretty(entry.getValue()));
```

Comment:
Here is the note from discussion with @webron:
> If the user provided:
>"application/xml": {
>    "name": "Puma",
>    "type": "Dog",
>    "color": "Black",
>    "gender": "Female",
>    "breed": "Mixed"
>  }
>we are going to show it just like that. A JSON.
>If they want an XML there, let them put an XML inside a string.
So, the proposed changes provide exactly this functionality.